torch/utils/_triton.py

```diff
@@ -1,14 +1,21 @@
 # mypy: allow-untyped-defs
 import functools
 import hashlib
+import importlib
 
 
 @functools.lru_cache(None)
 def has_triton_package() -> bool:
     try:
-        from triton.compiler.compiler import triton_key
-
-        return triton_key is not None
+        import triton
+        if triton.language.dtype is not None:
+            try:
+                from triton.compiler.compiler import triton_key
+                return triton_key is not None
+            except ImportError:
+                return False
+    except AttributeError:
+        return False
```

Comment:
This should be enough, not sure why you need to add the dtype stuff

```suggestion
        import triton
        from triton.compiler.compiler import triton_key
        return triton_key is not None
    except AttributeError:
        return False
```

Reply:
The main situation here is a pytorch environment without tirton installed. Then a folder which has the same name with triton will cause init issue due to the way of import search in torch 2.4